pandas/tests/series/test_missing.py

```diff
@@ -1398,9 +1398,21 @@ def test_nonzero_warning(self):
         with tm.assert_produces_warning(FutureWarning):
             ser.nonzero()
 
-    def test_interpolate_non_numeric_index(self, nontemporal_method):
+    @pytest.mark.parametrize(
+        "ind",
+        [
+            ['a', 'b', 'c', 'd'],
+            pd.period_range(start="2019-01-01", periods=4),
+            pd.timedelta_range(start=1, periods=4),
+            pd.interval_range(start=0, end=4),
+        ])
```

Comment:
@jreback, parameterized the test on different indexes.
Note that, ```TimedeltaIndex``` currently supports `time`, `index`, `values` and `pchip` methods of interpolation. In generic.py because of the check of numeric and datetime64 types, even for these methods it will raise the ValueError.

Should I change the if condition to contain the timedelta and period and interval types as well and create a test for each index to xfail? in case of timedelta then some will pass and some will xfail
or
write an if else logic in the generic.py to cover those timedelta cases?

Reply:
ideally this PR is just changing the error as the original issue + tests (xfailednif need be for anything else)

if after that is passing you want to try to remove the xfails would be ok

but let’s get the tests correct first